### PR TITLE
Updating to embedded-hal digital::v2

### PR DESCRIPTION
**Pull request by careyk007**
I have updated the GPIO pins to use the embedded-hal v2 traits. I only have an Adafruit nRF52832 board so I haven't been able to test other chips, but since most of the changes were in `nrf52-hal-common` I am confident that this will work for other chips as well.

### CODE REVIEW

File: nrf52-hal-common/src/gpio.rs [hunk 1]
```diff
@@ -62,7 +62,8 @@ use crate::target::P0;
 #[cfg(feature = "52840")]
 use crate::target::{ P1 };
 
-use crate::hal::digital::{OutputPin, StatefulOutputPin, InputPin};
+use crate::hal::digital::v2::{OutputPin, StatefulOutputPin, InputPin};
+use core::convert::Infallible;
```
Comment (thejpster) [inline nrf52-hal-common/src/gpio.rs]: Does using this increase our MSRV? IIRC this came in in 1.34.

Comment (hannobraun) [reply nrf52-hal-common/src/gpio.rs]: I don't know what the MSRV is, but if `Infallible` needs to be replaced, it should be replaced with `void::Void`, which is already a dependency of this crate.

Comment (careyk007) [reply nrf52-hal-common/src/gpio.rs]: Awesome, I went ahead and replaced `Infallible` with `void::Void`.

**Comment by hannobraun** [comment]
Thank you for the pull request, @careyk007!

I have one more nit-pick: You added `embedded-hal` as a direct dependency to some crates, to import the new traits from there. I think it would be better to add the new traits to the prelude (`nrf52-hal-common/lib.rs`) and import them through that.

Other than that, I think this is good to merge!

**Comment by hannobraun** [comment]
Thanks for the changes! Travis is failing. I think you're missing an import somewhere.

Also, I was suggesting to re-export the `digital::v2` traits from the `prelude` module in `lib.rs`, but that can also be done in a later step. I think this is good to merge as soon as the CI build is green.

**Comment by hannobraun** [comment]
Looks good to, thanks!

Also see https://github.com/nrf-rs/nrf52-hal/pull/117#issuecomment-543518176.

**Pull request merged**